### PR TITLE
Fix schema.VerifyPassword type assertion

### PR DESCRIPTION
**Pull request by markostojanovic087**
Fixing issue [Issue-243](https://github.com/rs/rest-layer/issues/243)

### CODE REVIEW

**Comment by markostojanovic087** [comment]
I think now that examples provided will fail since they have this code:
```
{ID: "admin", Updated: time.Now(), ETag: "abcd", Payload: map[string]interface{}{
			"id":       "admin",
			"name":     "Dilbert",
			"password": secret,
		}},
```
As you can see the password isn't quoted. I think we should change it also.
@smyrman Do you agreee?

By the way, there is another mistake in this examples which makes some of it's code useless.

Please see the [line](https://github.com/rs/rest-layer/blob/3f2cd5df9e8e6690c56f806710ef53048109c323/examples/auth/main.go#L95).
The correct line will be 
```
if err != nil {
```

The way it is, the code always returns.
@smyrman Should we open another issue or make this change along with this one?

File: schema/password.go [hunk 1]
```diff
@@ -58,9 +58,9 @@ func (v Password) Validate(value interface{}) (interface{}, error) {
 // VerifyPassword compare a field of an item payload containing a hashed
 // password with a clear text password and return true if they match.
 func VerifyPassword(hash interface{}, password []byte) bool {
-	h, ok := hash.([]byte)
+	h, ok := hash.(string)
 	if !ok {
 		return false
 	}
-	return bcrypt.CompareHashAndPassword(h, password) == nil
```
Comment (smyrman) [inline schema/password.go]: I want to suggest this instead:
```go
var h []byte
switch hashT := hash.(type) {
case []byte:
    b = hashT
case string:
    b = []byte(hashT)
default:
    return false
}
return bcrypt.CompareHashAndPassword(h, password) == nil
```

Then we handle both a `[]byte` and a `string` DB value, which both probably can be the case, depending on schema and/or backend.

**Comment by smyrman** [comment]
> @smyrman Should we open another issue or make this change along with this one?

Maybe we can do a separate PR to update any issues in the example / README, then this can be merged sooner.

**Comment by smyrman** [comment]
Closing due to lack of activity.

**Pull request closed without merging**